### ISSUE TITLE
D103 wrongly reported on function with dict type hint

### ISSUE THREAD

**Issue by syntaxaire** [PyCQA/flake8-docstrings#41]
Given this file:

```python
"""Sample file."""


def get_map(modules: list) -> dict[str:dict]:
    """
    A docstring.

    :param modules: A list
    :return: A dictionary mapping strings to dictionaries.
    """
    print(modules)
    return {"A": {"A": 1}, "B": {"B": 2}}
```

flake8-docstrings will report:
```
.\file.py:4:1: D103 Missing docstring in public function
```

Reproducible with Python 3.10.4, flake8 4.0.1, and flake8-docstrings 1.6.0. Removing the return type hint stops the false positive.

**Comment by asottile** [comment]
flake8-docstrings does not implement any checks -- please report to pydocstyle

note that the problem is likely because your return annotation is a typo, you want `dict[str, dict]`

**Comment by syntaxaire** [comment]
Thanks!
